Source/WebKit/UIProcess/glib/SystemSettingsManager.cpp

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright (C) 2021 Purism SPC
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE INC. OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+#include "SystemSettingsManager.h"
+
+#include "SystemSettingsProxyMessages.h"
+#include "WebProcessPool.h"
+
+namespace WebKit {
+using namespace WebCore;
+
+#if !PLATFORM(GTK)
+
+SystemSettingsManager::SystemSettingsManager() = default;
+
+String SystemSettingsManager::themeName() const
+{
+    return emptyString();
+}
+
+String SystemSettingsManager::fontName() const
+{
+    return "Sans 10"_s;
+}
+
+int SystemSettingsManager::xftAntialias() const
+{
+    return -1;
+}
+
+int SystemSettingsManager::xftHinting() const
+{
+    return -1;
+}
+
+String SystemSettingsManager::xftHintStyle() const
+{
+    return emptyString();
+}
+
+String SystemSettingsManager::xftRGBA() const
+{
+    return "rgb"_s;
+}
+
+int SystemSettingsManager::xftDPI() const
+{
+    return -1;
+}
+
+bool SystemSettingsManager::cursorBlink() const
+{
+    return true;
+}
+
+int SystemSettingsManager::cursorBlinkTime() const
+{
+    return 1200;
+}
+
+bool SystemSettingsManager::primaryButtonWarpsSlider() const
+{
+    return true;
+}
+
+bool SystemSettingsManager::overlayScrolling() const
+{
+    return true;
+}
+
+bool SystemSettingsManager::enableAnimations() const
+{
+    return true;
+}
+
+#endif // !PLATFORM(GTK)
+
+void SystemSettingsManager::initialize()
+{
+    static NeverDestroyed<SystemSettingsManager> manager;
+}
```

Comment:
I don't understand this

Reply:
The manager has to be created after GtkSettings is but before anything uses it, so we have a method to create it. This function just creates it on first use.

Previously the manager just needed to exist before the webprocess launched, as the UI process didn't rely on it.

Reply:
Is that the case being a static var? Don't you need to use LazyNeverDestroyed for that to work?

Reply:
In testing it definitely worked but `LazyNeverDestroyed` sounds like a perfect fit so I'll switch to it, maybe it handles some edge case.

Reply:
`LazyNeverDestroyed` lets you construct a static variable in a thread-safe way. A static `NeverDestroyed` is just immediately constructed when the function is called, but only once. So i think this is fine.